language-family/scripts/package.sh

```diff
@@ -0,0 +1,191 @@
+      ;;
+
+    *)
+      echo "Unknown OS \"$(uname)\""
+      exit 1
+  esac
+
+  readonly OS="${os}"
+}
+
+function repo::prepare() {
+  echo "-> Preparing repo..."
+
+  rm -rf "${BUILD_DIR}"
+
+  mkdir -p "${BIN_DIR}"
+  mkdir -p "${BUILD_DIR}"
+
+  export PATH="${BIN_DIR}:${PATH}"
+}
+
+function tools::jam::install() {
+  echo "-> Installing latest jam..."
+
+  local os
+  os="${OS}"
+  if [[ "${os}" == "macos" ]]; then
+    os="darwin"
+  fi
+
+  local url
+  url="$(
+    curl "https://api.github.com/repos/cloudfoundry/packit/releases/latest" \
+      --header "Authorization: token ${GITHUB_TOKEN}" \
+      --silent \
+      --location \
+      | jq -r ".assets[] | select(.name | contains(\"${os}\")) | .browser_download_url"
+  )"
+
+  curl "${url}" \
+    --silent \
+    --location \
+    --output "${BIN_DIR}/jam"
+  chmod +x "${BIN_DIR}/jam"
+}
+
+function tools::yj::install() {
+  echo "-> Installing latest yj..."
+
+  local url
+  url="$(
+    curl "https://api.github.com/repos/sclevine/yj/releases/latest" \
+      --header "Authorization: token ${GITHUB_TOKEN}" \
+      --silent \
+      --location \
+      | jq -r ".assets[] | select(.name | contains(\"${OS}\")) | .browser_download_url"
+  )"
+
+  curl "${url}" \
+    --silent \
+    --location \
+    --output "${BIN_DIR}/yj"
+  chmod +x "${BIN_DIR}/yj"
+}
+
+function tools::pack::install() {
+  echo "-> Installing latest pack..."
+
+  local url
+  url="$(
+    curl "https://api.github.com/repos/buildpacks/pack/releases/latest" \
+      --header "Authorization: token ${GITHUB_TOKEN}" \
+      --silent \
+      --location \
+      | jq -r ".assets[] | select(.name | contains(\"${OS}\")) | .browser_download_url"
+  )"
+
+  curl "${url}" \
+    --silent \
+    --location \
+    --output /tmp/pack.tgz
+  tar xzf /tmp/pack.tgz -C "${BIN_DIR}"
+  chmod +x "${BIN_DIR}/pack"
+  rm /tmp/pack.tgz
+}
+
+function buildpack::archive() {
+  local version
+  version="${1}"
+
+  echo "-> Packaging family buildpack into ${BUILD_DIR}/buildpack.tgz..."
+
+  jam pack \
+    --buildpack "${ROOT_DIR}/buildpack.toml" \
+    --version "${version}" \
+    --output "${BUILD_DIR}/buildpack.tgz"
+}
+
+function buildpackage::toml::write() {
+  echo "-> Generating package config in ${BUILD_DIR}/package.toml..."
+
+  yj -tj < "${ROOT_DIR}/buildpack.toml" \
+    | jq -r '.metadata.dependencies[] | select(.id != "lifecycle") | {uri: .uri }' \
+    | jq -s --arg uri "${BUILD_DIR}/buildpack.tgz" '. | {buildpack: {uri: $uri}, dependencies: .}' \
+    | yj -jt \
+    > "${BUILD_DIR}/package.toml"
+}
+
+function buildpackage::create() {
+  echo "-> Packaging buildpack..."
+
+  pack \
+    package-buildpack "${BUILD_DIR}/buildpackage.cnb" \
+      --package-config "${BUILD_DIR}/package.toml" \
+      --format file
+}
+
+function github::token::fetch() {
+  if [[ -z "${GITHUB_TOKEN:-""}" ]]; then
+    echo "-> Fetching GITHUB_TOKEN..."
+
+    GITHUB_TOKEN="$(
+      lpass show Shared-CF\ Buildpacks/concourse-private.yml \
```

Comment:
Would it be better to hard-code versions of jam, pack etc now so that you don't need lpass private bucket access, and what's tested in a PR will be the same as the one in the base branch?

Reply:
That seems reasonable. We will change this.